gherkin/php/tests/Parser/RuleTypeTest.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Cucumber\Gherkin\Parser;
+
+use Generator;
+use PHPUnit\Framework\TestCase;
+
+final class RuleTypeTest extends TestCase
+{
+    /** @dataProvider tokenCaseProvider */
+    public function testItCanCastFromTokens(TokenType $tokenType): void
+    {
+        self::assertInstanceOf(RuleType::class, RuleType::cast($tokenType));
+    }
+
+    /**
+     * @return Generator<string,array{0:TokenType}>
+     */
+    public function tokenCaseProvider(): Generator
```

Comment:
I highly recommend to make PHPUnit data providers `static`. This avoids ever trying to use `$this` in it which would be a nasty surprise (the instance of the test case inside the provider is different than the one executing the test)